server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4336,6 +4336,18 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                 vm.setUserVmType(type);
                 _vmDao.persist(vm);
                 for (String key : customParameters.keySet()) {
+                    // BIOS was explicitly passed as the boot type, so honour it
```

Comment:
what if template details has UEFI -> LEGACY and deploy VM passes UEFI -> SECURE ? As per my understanding, we are considering template details, since we are not deleting from vm.details if boot type is UEFI. (vn.details is loaded with template details before this).

If precedence goes to template details this is fine, but it might be confusing to the user if he/she provides SECURE mode during VM deploy but it is not honoured since template has other detail.

Reply:
The template boot mode will be overwritten by the boot mode passed in the deployvm command in line 4345
Precedence is as follows : deploy-as-is > deployvm > VM details > template details

Reply:
got it, thanks David